web/src/HUD.tsx

```diff
@@ -180,6 +188,35 @@ class HUD extends Component<HudProps, HudState> {
       .catch(err => console.error(err))
   }
 
+  highlightsEnabled(): boolean {
+    return (
+      !this.pathBuilder.isSnapshot() &&
+      this.getFeatures().isEnabled("snapshot_highlights")
+    )
+  }
+
+  private getFeatures(): Features {
+    if (this.state.View) {
+      this.features = new Features(this.state.View.FeatureFlags)
```

Comment:
i don't understand why we're storing this as a property of the class at all?

currently, we write the field every time we read the field

Reply:
Whoops, I meant to memoize this but messed up. Thanks

Reply:
In the end I decided not to bother memoizing it.